src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/DF_BaseRecalibrator.java

```diff
@@ -76,68 +84,134 @@
      */
     private String beforePath = null;
 
-
     /**
      * Path to the reference (SAM file). Eventually can be filesystem or gs://
      */
     private String referencePath = null;
 
     /**
-     * Path to save the output to. Must be local for now.
+     * Path to save the serialized tables to. Local or GCS.
      */
-    @Argument(doc = "Path to save the output to. Must be filesystem.", optional = false)
-    protected String outputTablesPath = "df-recaltables.sj";
+    @Argument(doc = "Path to save the serialized recalibrationTables to.", optional = true)
+    protected String outputTablesPath = null;
 
-    // ------------------------------------------
-    // Dataflow-specific options
+    // Whether we run on the cloud.
+    private boolean remote = false;
 
-    @Argument(doc = "path to client-secrets.json", optional = false)
-    protected String secretsFile;
+    // the inputs to BQSR
+    private PCollection<Read> reads;
+    private PCollection<SimpleInterval> skipIntervals;
+    private SAMFileHeader header;
 
-    // not actually optional if you ask for remote exec
-    @Argument(doc = "Google Cloud project name", optional = true)
-    protected String project;
+    // set up with the user's arguments, kept around to save the textual report.
+    private BaseRecalibratorUprooted baseRecalibratorUprooted;
 
-    // not actually optional if you ask for remote exec
-    @Argument(doc = "Google Cloud Storage path for intermediate files", optional = true)
-    protected String stagingLocation;
+    @Override
+    protected void setupPipeline(Pipeline pipeline) {
+        try {
+            remote = runnerType != PipelineRunnerType.LOCAL;
+            ingestInputs(pipeline);
+            BQSR_Dataflow.ensureReferenceIsReadable(pipeline.getOptions(), referencePath);
+            baseRecalibratorUprooted = BaseRecalibratorUprooted.fromArgs(header, BRAC);
+            baseRecalibratorUprooted.checkClientArguments();
 
-    @Argument(doc = "Number of Dataflow workers", optional = true)
-    protected int numWorkers = 0;
+            // 2. set up computation
+            PCollection<RecalibrationTables> aggregated =
+                    BQSR_Dataflow.GetRecalibrationTables(header, reads, referencePath, BRAC, skipIntervals);
 
-    // ------------------------------------------
+            if (null == outputTablesPath) {
+                // we need those, so let's pick a location for them.
+                if (remote) {
+                    outputTablesPath = GcsPath.fromUri(stagingLocation).resolve("temp-ds-recaltables.sj").toString();
+                } else {
+                    File outputTables = File.createTempFile("temp-ds-recaltables", ".sj");
+                    outputTables.deleteOnExit();
```

Comment:
This won't compile. I think that the problem is that test classes are typically not included when executing (or compiling) the main program.